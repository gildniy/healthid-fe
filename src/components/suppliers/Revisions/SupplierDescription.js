import React from 'react';
import {
  Grid, Card, CardMedia
} from '@material-ui/core';
import PropTypes from 'prop-types';
import Rating from '../Templates/Rating';
import { FieldTitle, getFields } from './SupplierFields';


const SupplierDescription = (props) => {
  const {
    classes, supplier, currentSupplier, activeOutlet
  } = props;
  const outletId = activeOutlet.id;
  const { supplierratingSet, supplierContacts: supplierscontactsSet } = currentSupplier;
  const { supplierContacts, supplierMeta } = supplier;
  const rating = supplierratingSet[0] && supplierratingSet[0].rating;
  const contacts = supplierContacts[0];
  const currentContacts = supplierscontactsSet.find(contact => contact.outlet.id === outletId);
  const FieldValues = getFields(currentContacts, contacts, classes);
  const { fieldsChanged } = FieldValues;
  const meta = supplierMeta[0];

  return (
    <>
      <Grid
        container
        spacing={24}
        className={classes.containerGrid}
      >
        <Grid container item xs={12}>
          <Grid container item xs={8} spacing={24}>
            <Grid item xs={12}>
              <div style={{ marginBottom: '30px' }} className={classes.category}>
                <FieldTitle changed title="Supplier ID" classes={classes} />
                <div style={{ marginTop: '5px' }}>
                  <span className={classes.addressText}>{supplier.id}</span>
                </div>
              </div>
              <div>
                <FieldTitle
                  changed={['addressLine1', 'addressLine2', 'lga', 'country', 'city']
                    .some(key => Object.keys(fieldsChanged).includes(key))}
                  title="Address"
                  classes={classes}
                />
                <FieldValues.Render name="addressLine1" />
                <FieldValues.Render name="addressLine2" />
                <FieldValues.Render name="lga" />
                <FieldValues.Render name="city_country" />
              </div>
            </Grid>
          </Grid>
          <Grid item xs={4}>
            <Card elevation={0} className={classes.card}>
              <CardMedia
                className={classes.media}
                image={meta && meta.logo}
                title="Supplier Image"
              />
            </Card>
          </Grid>
        </Grid>
      </Grid>
      <Grid container spacing={24} className={classes.containerGrid}>
        <Grid item xs={4}>
          <FieldTitle changed={'email' in fieldsChanged} title="Email" classes={classes} />
          <FieldValues.Render name="email" />
        </Grid>

        <Grid item xs={4}>
          <FieldTitle changed={'mobileNumber' in fieldsChanged} title="Mobile #" classes={classes} />
          <FieldValues.Render name="mobileNumber" />
        </Grid>

        <Grid container item xs={4}>
          <Rating rating={rating} starClass="supplierStar" />
        </Grid>
      </Grid>
    </>
  );
};

SupplierDescription.propTypes = {
  classes: PropTypes.shape({
    containerGrid: PropTypes.string,
    containerGrid2: PropTypes.string,
    category: PropTypes.string,
    card: PropTypes.string,
    media: PropTypes.string,
    descriptionFields: PropTypes.string,
    addressText: PropTypes.string,
    addressTextHeader: PropTypes.string
  }).isRequired,
  supplier: PropTypes.instanceOf(Object).isRequired,
  currentSupplier: PropTypes.instanceOf(Object).isRequired,
  activeOutlet: PropTypes.instanceOf(Object).isRequired
};

export default SupplierDescription;
