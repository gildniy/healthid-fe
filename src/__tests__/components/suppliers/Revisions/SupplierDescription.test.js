import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SupplierDescription from '../../../../components/suppliers/Revisions/SupplierDescription';

const props = {
  supplier: {
    isApproved: false,
    id: 'AlphaX',
    address: '​​​​​NNPC Towers, Central Business District, Abuja.',
    name: 'Damian Inc',
    mobile: '08137519698',
    rating: '5',
    image:
      'https://justcreative.com/wp-content/uploads/2019/09/brand-strategy-workbook-1.jpg',
    email: 'test@gmail.com',
    tier: 'Manufacturer',
    paymentTerms: 'Daily',
    supplierratingSet: [],
    supplierMeta: [{
      logo: 'x'
    }],
    creditDays: '3 days',
    supplierContacts: [
      {
        addressLin1: '144 Peter Road',
        addressLine2: 'JanBleck 20',
        city: {
          name: 'Kampala'
        },
        country: {
          name: 'Uganda'
        }
      },
    ],
  },
  currentSupplier: {
    isApproved: true,
    id: 'AlphaX',
    address: '​​​​​NNPC Towers, Central Business District, Abuja.',
    name: 'Damian Inc',
    mobile: '08137519698',
    rating: '5',
    image:
      'https://justcreative.com/wp-content/uploads/2019/09/brand-strategy-workbook-1.jpg',
    email: 'test@gmail.com',
    tier: 'Manufacturer',
    paymentTerms: 'Daily',
    supplierratingSet: [{
      rating: 5
    }],
    supplierMeta: [{
      logo: 'x'
    }],
    creditDays: '3 days',
    supplierContacts: [
      {
        addressLin1: '144 Peter Road',
        addressLine2: 'JanBleck 20',
        city: {
          name: 'Kampala'
        },
        country: {
          name: 'Uganda'
        }
      },
    ],
  },
  classes: {},
  session: {
    me: {
      role: {
        name: 'Master Admin'
      },
      activeOutlet: {
        outletpreference: {
          outletTimezone: {
            name: 'Africa/Nairobi'
          }
        }
      }
    }
  }
};

describe('SupplierDescription', () => {
  it('should render the SupplierDescription component properly', () => {
    const setupSupplierDescription = () => {
      const wrapper = shallow(<SupplierDescription {...props} />);
      return wrapper;
    };

    const wrapper = setupSupplierDescription();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
