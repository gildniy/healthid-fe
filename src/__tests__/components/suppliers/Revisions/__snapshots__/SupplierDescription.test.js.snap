// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SupplierDescription should render the SupplierDescription component properly 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={24}
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      item={true}
      xs={12}
    >
      <WithStyles(ForwardRef(Grid))
        container={true}
        item={true}
        spacing={24}
        xs={8}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={12}
        >
          <div
            style={
              Object {
                "marginBottom": "30px",
              }
            }
          >
            <FieldTitle
              changed={true}
              classes={Object {}}
              title="Supplier ID"
            />
            <div
              style={
                Object {
                  "marginTop": "5px",
                }
              }
            >
              <span>
                AlphaX
              </span>
            </div>
          </div>
          <div>
            <FieldTitle
              changed={false}
              classes={Object {}}
              title="Address"
            />
            <Render
              name="addressLine1"
            />
            <Render
              name="addressLine2"
            />
            <Render
              name="lga"
            />
            <Render
              name="city_country"
            />
          </div>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={4}
      >
        <WithStyles(ForwardRef(Card))
          elevation={0}
        >
          <WithStyles(ForwardRef(CardMedia))
            image="x"
            title="Supplier Image"
          />
        </WithStyles(ForwardRef(Card))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={24}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={4}
    >
      <FieldTitle
        changed={false}
        classes={Object {}}
        title="Email"
      />
      <Render
        name="email"
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={4}
    >
      <FieldTitle
        changed={false}
        classes={Object {}}
        title="Mobile #"
      />
      <Render
        name="mobileNumber"
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      container={true}
      item={true}
      xs={4}
    >
      <Rating
        rating={5}
        starClass="supplierStar"
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</Fragment>
`;
